### PR TITLE
Unify usage of custom HTTP clients

### PR DESCRIPTION
**Pull request by adriansmares**
#### Summary

References https://github.com/TheThingsNetwork/lorawan-stack/issues/2533
References https://github.com/TheThingsNetwork/lorawan-stack/pull/3486

#### Changes

- Remove the LoRaCloud Geolocation custom `*http.Client`
- `fetch.FromHTTP` now takes a `*http.Client` instead of `http.RoundTripper`
  - This allows the caller to control the request timeouts (and removes redudant magic constants from our code base)
- Adapt HTTP fetcher configuration to use `*http.Client` instead of `http.RoundTripper`
  - Originally these configs contained a `http.RoundTripper` field which was used by `fetch.FromHTTP`
- AWS Blob access now uses a custom `*http.Client`
  - This is part of an ongoing effort to use timeouts for all of the operations that contact external services, including cloud provider services.
- The interop client no longer uses a custom `http.Client`.
  - The risk here was twofold - the transport used before didn't contain any timeouts, and the `http.Client` itself did not have a timeout.

#### Testing

Unit testing and :monkey: testing.

##### Regressions

Several HTTP fetcher initialization sites are now updated. Nonetheless, the changes are identical, and I've tested the webhook template fetcher (visit the webhook creation screen in the Console).

#### Checklist

- [x] Scope: The referenced issue is addressed, there are no unrelated changes.
- [x] Compatibility: The changes are backwards compatible with existing API, storage, configuration and CLI, according to the compatibility commitments in `README.md` for the chosen target branch.
- [ ] Documentation: Relevant documentation is added or updated.
- [ ] Changelog: Significant features, behavior changes, deprecations and fixes are added to `CHANGELOG.md`.
- [x] Commits: Commit messages follow guidelines in `CONTRIBUTING.md`, there are no fixup commits left.

### CODE REVIEW

**Review by htdvisser — APPROVED** [review]

**Review by KrishnaIyer — APPROVED** [review]
Code owned files LGTM

File: cmd/ttn-lw-stack/commands/start.go [hunk 1]
```diff
@@ -190,15 +190,21 @@ var startCommand = &cobra.Command{
 
 		redisConsumerID := redis.Key(host, strconv.Itoa(os.Getpid()))
 
-		transport, err := c.HTTPTransport(ctx)
-		if err != nil {
-			return err
-		}
-		if conf := config.ServiceBase.FrequencyPlans; conf.Transport == nil {
```
Comment (adriansmares) [inline cmd/ttn-lw-stack/commands/start.go:197]: These two fillers actually did nothing since `conf` is not a pointer.

**Review by kschiffer — APPROVED** [review]
Codeowned LGTM. E2E fail is a flake, I've restarted the job.

**Comment by adriansmares** [comment]
@johanstokking / @rvolosatovs please review the code owned files (I believe only `pkg/interop` is left). Thanks.

**Review by johanstokking — APPROVED** [review]

**Pull request merged**